iotdb/src/main/java/org/apache/iotdb/db/sync/sender/FileSenderImpl.java

```diff
@@ -0,0 +1,507 @@
+    Date oldTime = new Date();
+    while (!Thread.interrupted()) {
+      Date currentTime = new Date();
+      if (currentTime.getTime() / 1000 == oldTime.getTime() / 1000) {
+        continue;
+      }
+      if ((currentTime.getTime() - lastSyncTime.getTime())
+          % (config.getUploadCycleInSeconds() * 1000) == 0) {
+        oldTime = currentTime;
+        if (syncStatus) {
+          LOGGER.info("Sync process is in execution!");
+        }
+      }
+    }
+  };
+
+  private FileSenderImpl() {
+  }
+
+  public static final FileSenderImpl getInstance() {
+    return InstanceHolder.INSTANCE;
+  }
+
+  /**
+   * Create a sender and sync files to the receiver.
+   *
+   * @param args not used
+   */
+  public static void main(String[] args)
+      throws InterruptedException, IOException, SyncConnectionException {
+    Thread.currentThread().setName(ThreadName.SYNC_CLIENT.getName());
+    FileSenderImpl fileSenderImpl = new FileSenderImpl();
+    fileSenderImpl.verifySingleton();
+    fileSenderImpl.startMonitor();
+    fileSenderImpl.timedTask();
+  }
+
+  /**
+   * Start Monitor Thread, monitor sync status
+   */
+  public void startMonitor() {
+    Thread syncMonitor = new Thread(monitorSyncStatus, ThreadName.SYNC_MONITOR.getName());
+    syncMonitor.setDaemon(true);
+    syncMonitor.start();
+  }
+
+  /**
+   * Start sync task in a certain time.
+   */
+  public void timedTask() throws InterruptedException, SyncConnectionException, IOException {
+    sync();
+    lastSyncTime = new Date();
+    Date currentTime;
+    while (true) {
+      if (Thread.interrupted()) {
+        break;
+      }
+      Thread.sleep(2000);
+      currentTime = new Date();
+      if (currentTime.getTime() - lastSyncTime.getTime()
+          > config.getUploadCycleInSeconds() * 1000) {
+        lastSyncTime = currentTime;
+        sync();
+      }
+    }
+  }
+
+  /**
+   * Execute a sync task.
+   */
+  @Override
+  public void sync() throws SyncConnectionException, IOException {
+
+    //1. Clear old snapshots if necessary
+    for (String snapshotPath : config.getSnapshotPaths()) {
+      if (new File(snapshotPath).exists() && new File(snapshotPath).list().length != 0) {
+        /** It means that the last task of sync does not succeed! Clear the files and start to sync again **/
+        SyncUtils.deleteFile(new File(snapshotPath));
+      }
+    }
+
+    // 2. Acquire valid files and check
+    fileManager.init();
+    validAllFiles = fileManager.getValidAllFiles();
+    currentLocalFiles = fileManager.getCurrentLocalFiles();
+    if (SyncUtils.isEmpty(validAllFiles)) {
+      LOGGER.info("There has no file to sync !");
+      return;
+    }
+
+    // 3. Connect to sync server and Confirm Identity
+    establishConnection(config.getServerIp(), config.getServerPort());
+    if (!confirmIdentity(config.getUuidPath())) {
+      LOGGER.error("Sorry, you do not have the permission to connect to sync receiver.");
+      return;
```

Comment:
I think in this case, you should just exit the sync thread and report a fatal error because the same error will occur the next time sync() is called.

Reply:
It sounds reasonable, fixed